pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -95,3 +96,82 @@ func createEcNegateHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 
 	return newEcNegateHint(point), nil
 }
+
+func newNondetBigint3V1Hint(res hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "NondetBigint3V1",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import split
+			//> segments.write_arg(ids.res.address_, split(value))
+
+			address, err := res.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			value, err := ctx.ScopeManager.GetVariableValue("value")
+			if err != nil {
+				return err
+			}
+			valueBig, ok := value.(*big.Int)
+			if !ok {
+				return fmt.Errorf("value: %s is not a *big.Int", value)
+			}
+
+			baseBig, ok := utils.GetEcBaseBig()
+			if !ok {
+				return fmt.Errorf("GetEcBaseBig failed")
+			}
+
+			//> def split(num: int) -> List[int]:
+			//>     """
+			//?     Takes a 256-bit integer and returns its canonical representation as:
+			//>         d0 + BASE * d1 + BASE**2 * d2,
+			//>     where BASE = 2**86.
+			//>     """
+			//?     a = []
+			//>     for _ in range(3):
+			//>         num, residue = divmod(num, BASE)
+			//>         a.append(residue)
+			//>     assert num == 0
+			//>     return a
+			//>
+			//> segments.write_arg(ids.res.address_, split(value))
+
+			var residue big.Int
```

Comment:
we should move those functionalities to the separate secp utils file. Maybe the generation of the constants as well?

Reply:
I mean the secp constants

Reply:
Yeah lots of functionality is clumped in 1 utils file right now. We need separate files for math utils, secp utils, etc.

Reply:
I think its cleaner to just write a `split` function and use it here. I'll do that 👍🏽

Reply:
Done! Please check.

Reply:
Looks good!